src/proto/extract.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS=$'\n\t'
+
+# --- begin runfiles.bash initialization v3 ---
+# Copy-pasted from the Bazel Bash runfiles library v3.
```

Comment:
The Bazel Bash runfiles library v3 is subject to [the Apache License v2](https://www.apache.org/licenses/LICENSE-2.0), which requires, among other things, a copy of the license and retention of copyright notices. I see that there is a copyright notice at the top of https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash#L1

Why does this have to be copy-pasted rather than used?

Reply:
We can proceed with this PR, as I look over whether we could handle this differently 👍